<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1411</width>
    <height>1079</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="workArea">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>731</width>
      <height>631</height>
     </rect>
    </property>
    <property name="frameShape">
     <enum>QFrame::Shape::Box</enum>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>160</y>
      <width>251</width>
      <height>291</height>
     </rect>
    </property>
    <property name="title">
     <string>CIRCLE</string>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>29</y>
       <width>231</width>
       <height>251</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QLabel" name="radiuslabel">
        <property name="font">
         <font>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Radius</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDoubleSpinBox" name="Radius"/>
      </item>
      <item>
       <widget class="QPushButton" name="GenMidPtCircle">
        <property name="text">
         <string>Generate Circle MidPoint</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="GenCirclePolar">
        <property name="text">
         <string>Generate Circle Polar</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="GenBresenhamCircle">
        <property name="text">
         <string>Generate Circle Bresenham</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="GenCartCircle">
        <property name="text">
         <string>Generate Cartesian Circle</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>160</y>
      <width>281</width>
      <height>281</height>
     </rect>
    </property>
    <property name="title">
     <string>Ellipse</string>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>30</x>
       <y>40</y>
       <width>195</width>
       <height>187</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QDoubleSpinBox" name="minor"/>
      </item>
      <item>
       <widget class="QLabel" name="label_5">
        <property name="font">
         <font>
          <pointsize>9</pointsize>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Semi Minor Axis</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QDoubleSpinBox" name="major"/>
      </item>
      <item>
       <widget class="QLabel" name="label_6">
        <property name="font">
         <font>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Semi Major Axis</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="PolarEllipse">
        <property name="text">
         <string>Generate Ellipse Polar</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="GenEllipseBre">
        <property name="text">
         <string>Generate Ellipse Bresenham</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>60</y>
      <width>460</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="showAxis">
       <property name="text">
        <string>Show Axes</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Axis Width</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="axisWidth">
       <property name="minimum">
        <number>1</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>GridOffset</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="gridOffset">
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="singleStep">
        <number>10</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="gridlines">
       <property name="text">
        <string>Show Gridlines</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>20</y>
      <width>498</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>X Coordinate</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="x_coordinate">
       <property name="minimumSize">
        <size>
         <width>50</width>
         <height>0</height>
        </size>
       </property>
       <property name="frameShape">
        <enum>QFrame::Shape::Box</enum>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Y Coordinate</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="y_coordinate">
       <property name="minimumSize">
        <size>
         <width>50</width>
         <height>0</height>
        </size>
       </property>
       <property name="frameShape">
        <enum>QFrame::Shape::Box</enum>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="store">
       <property name="text">
        <string>Store Canvas</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="reset">
       <property name="text">
        <string>Erase Canvas</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>760</x>
      <y>100</y>
      <width>536</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QPushButton" name="BresenhamLineDraw">
       <property name="text">
        <string>Generate Bresenham Line</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="DDA_TIME">
       <property name="minimumSize">
        <size>
         <width>100</width>
         <height>0</height>
        </size>
       </property>
       <property name="frameShape">
        <enum>QFrame::Shape::Box</enum>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="DDADraw">
       <property name="text">
        <string>Generate DDA Line</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="Bresenham_Time">
       <property name="minimumSize">
        <size>
         <width>100</width>
         <height>0</height>
        </size>
       </property>
       <property name="frameShape">
        <enum>QFrame::Shape::Box</enum>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="DrawPolygon">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>450</y>
      <width>131</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Draw Polygon</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="polygonSideCount">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>450</y>
      <width>42</width>
      <height>26</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="Scanline">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>490</y>
      <width>131</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Scanline Fill</string>
    </property>
   </widget>
   <widget class="QPushButton" name="FloodFill4">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>530</y>
      <width>131</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Flood Fill</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="FourNeighbour">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>530</y>
      <width>110</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>4 Neighbour</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="EightNeighbour">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>570</y>
      <width>110</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>8 Neighbour</string>
    </property>
   </widget>
   <widget class="QPushButton" name="boundaryFill">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>570</y>
      <width>131</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Boundary Fill</string>
    </property>
   </widget>
   <widget class="QPushButton" name="zoomOut">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>450</y>
      <width>93</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Zoom Out</string>
    </property>
   </widget>
   <widget class="QPushButton" name="zoomIn">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>490</y>
      <width>93</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Zoom In</string>
    </property>
   </widget>
   <widget class="QLCDNumber" name="timetaken">
    <property name="geometry">
     <rect>
      <x>970</x>
      <y>610</y>
      <width>191</width>
      <height>41</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="TimeMessage">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>660</y>
      <width>591</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Segoe UI Black</family>
      <pointsize>9</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>610</y>
      <width>191</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Time taken in nanoseconds:</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="ThickEdges">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>490</y>
      <width>111</width>
      <height>24</height>
     </rect>
    </property>
    <property name="text">
     <string>Thick Edges</string>
    </property>
   </widget>
   <widget class="QPushButton" name="ReflectX">
    <property name="geometry">
     <rect>
      <x>1140</x>
      <y>450</y>
      <width>141</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Reflect w.r.t. x-axis</string>
    </property>
   </widget>
   <widget class="QPushButton" name="ReflectY">
    <property name="geometry">
     <rect>
      <x>1140</x>
      <y>490</y>
      <width>131</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Reflect w.r.t y-axis</string>
    </property>
   </widget>
   <widget class="QPushButton" name="ReflectO">
    <property name="geometry">
     <rect>
      <x>1140</x>
      <y>530</y>
      <width>141</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Reflect w.r.t origin</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1411</width>
     <height>26</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
